pkg/util/log/zapcore.go

```diff
@@ -0,0 +1,94 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+package log
```

Comment:
I think `zapcore.go` and `zapencoder.go` would better fit in a new, separate package (maybe `pkg/util/log/zap`?), that would make for a cleaner interface with `pkg/util/log` with exported functions, and avoid adding zap as a dependency of `pkg/util/log`. WDYT?

Reply:
That makes sense to me, would we export the `tracecWithDepth`-like functions, then? (so, in your other comment, we would rename e.g. `tracecWithDepth` to `TracecStackDepth`?)

Reply:
Yes, exactly